Tensile/Tests/integration/test_integration.py

```diff
@@ -0,0 +1,127 @@
+# 4. Run client with small subset of GEMM problems
+# $ pytest Tensile/Tests -m integration --capture=tee-sys --builddir /tmp/pytest
+
+def getLogicFileDir(baseDir, schedule):
+
+  prefix = "https://github.com/ROCmSoftwarePlatform/rocBLAS/raw/develop/library/src/blas3/Tensile/Logic/asm_full/"
+  testData = {
+    "pre_checkin" : [
+      "arcturus_Cijk_Ailk_Bjlk_SB.yaml",
+      "vega20_Cijk_Ailk_Bjlk_SB.yaml",
+      "vega10_Cijk_Ailk_Bjlk_SB.yaml",
+      "arcturus_Cijk_Ailk_Bjlk_BBH.yaml",
+      "arcturus_Cijk_Ailk_Bjlk_HBH.yaml",
+      "vega20_Cijk_Ailk_Bjlk_HBH.yaml",
+      "vega10_Cijk_Ailk_Bjlk_HBH.yaml",
+      "hip_Cijk_Ailk_Bjlk_CB.yaml"
+    ],
+    "quick" : [
+      "arcturus_Cijk_Ailk_Bjlk_SB.yaml",
+      "vega20_Cijk_Ailk_Bjlk_SB.yaml",
+      "vega10_Cijk_Ailk_Bjlk_SB.yaml"
+    ]
+  }
+
+  files = [prefix + e for e in testData[schedule] ]
+  destDir = os.path.join(baseDir, "logic_yaml", schedule)
+  Common.ensurePath(destDir)
+
+  import glob
+  for f in glob.glob(destDir+"/*"):
+    print("file: ", f)
+    if os.path.basename(f) not in testData[schedule]:
+      print("removing file: ", f)
+      os.remove(f)
+
+  cmd = list()
+  cmd.append("wget")
+  cmd.extend(files)
+  cmd.extend(["-nc"])
+  cmd.extend(["-P", destDir])
+  ret = subprocess.run(cmd)
+  assert(ret.returncode==0)
+
+  # pytest.set_trace()
+  return destDir
+
+# Mock class
+class ProblemSizesMock:
+  def __init__(self, exactLogic):
+    problems = [probSolPair[0] for probSolPair in exactLogic]
+    self.problems = [Problem(prob) for prob in problems]
+
+@pytest.mark.parametrize("mergeFiles",        [False, True])
+@pytest.mark.parametrize("legacyComponents",  [False, True])
+@pytest.mark.parametrize("libraryFormat",     ["yaml", pytest.param("msgpack", marks=pytest.mark.xfail)])
+@pytest.mark.parametrize("testYamls",         ["quick", "pre_checkin"])
+def test_integration(useGlobalParameters, builddir, testYamls, mergeFiles, libraryFormat, legacyComponents):
+  if testYamls == "pre_checkin":
+    if not (mergeFiles == True and libraryFormat == "yaml"): pytest.skip("only run pre_checkin test in certain combo")
+
+  logicFileDir = getLogicFileDir(builddir, testYamls)
+  outputDir    = os.path.join(builddir, "lib")
+
+  # pytest.set_trace()
+  with useGlobalParameters(OutputPath=outputDir,
+                           WorkingPath=outputDir,
+                           MergeFiles=mergeFiles,
+                           LibraryFormat=libraryFormat,
+                           LegacyComponents=legacyComponents,
+                           CMakeBuildType="Debug"
+                           ):
+    Common.ensurePath(outputDir)
+
+    # Note: this step is needed for retrieving code object and tensile library yaml paths
+    # TODO Do away with this workaround. TensileCreateLibrary.py should take function arguments and return values directly
+    #      Refactor TensileCreateLibrary (not .py) to take CLI args instead and pass to underlying python script
```

Comment:
You've duplicated a lot of TensileCreateLibrary here.   Why not just use a subprocess to call the script?  That way we test the entire process.

Reply:
That duplicated code is now removed. Tensile will now always generate manifest file so no extra run with `GenerateManifestAndExit=OFF` is needed.

The `getBuildNewClientLibraryScript()` is adopted here because it already does the job of translating global param to TensileCreateLibrary input args.